pkg/kubelet/kubelet.go

```diff
@@ -2788,29 +2798,27 @@ func (kl *Kubelet) recordNodeStatusEvent(eventtype, event string) {
 
 func (kl *Kubelet) syncNetworkStatus() {
 	var err error
-	if kl.configureCBR0 {
-		if kl.flannelExperimentalOverlay {
-			podCIDR, err := kl.flannelHelper.Handshake()
-			if err != nil {
-				glog.Infof("Flannel server handshake failed %v", err)
-				return
-			}
-			kl.updatePodCIDR(podCIDR)
-		}
-		if err := ensureIPTablesMasqRule(kl.nonMasqueradeCIDR); err != nil {
-			err = fmt.Errorf("Error on adding ip table rules: %v", err)
-			glog.Error(err)
-			kl.runtimeState.setNetworkState(err)
+	if kl.flannelExperimentalOverlay {
```

Comment:
Could we consolidate all the network state initialization/syncing to this function? E.g., if pod CIDR is required for the network plugins, we can still check whether pod CIDR is set and then call `setNetworkState()`. What do you think?